### DISCUSSION TITLE
posthtml options for singleTags doesn't appear to be working

### DISCUSSION THREAD

**Discussion by bravecrayon** [maizzle/framework#766] [General]
- Maizzle Version: 4.0.2
- Node.js Version: 14.19.3

I'm following the [documentation](https://maizzle.com/docs/configuration/posthtml) for setting up `singleTags` for a `<custom>` element – I think exactly the same use case as in the docs. I feel like I must be doing something silly wrong, but maybe it's a bug?

In config.js, I've have:

```JS
module.exports = {
  build: {
    templates: {
      source: 'src/templates',
      destination: {
        path: 'build_local',
      },
      assets: {
        source: 'src/images',
        destination: 'images',
      },
    },
    posthtml: {
      options: {
        singleTags: ['custom']
      }
    }
  },
}
```

In `src/layouts/main.html`, my expection is it would take this and render it as-is, with tags closed:
```html
  <custom name="opencounter" type="tracking" />
  <custom name="usermatch" type="tracking" />
```

However, the result I have is:
```html
  <custom name="opencounter" type="tracking">
  <custom name="usermatch" type="tracking">
```

Thanks!

**Comment by bravecrayon** [comment]
Dug into this a bit and figured out adding `closingSingleTag` makes it work, but it does apply this closing format on all recognized single tags.

```js
posthtml: {
  options: {
    singleTags: ['custom'],
    closingSingleTag: 'slash'
  }
}
```

**Comment by cossssmin** [comment]
Sorry this took me a while to get to, I'll look into it but I do think that's how PostHTML works (requiring `closingSingleTag` and applying it to all known tags, not just those in `singleTags`).

**Comment by cossssmin** [comment]
Looked into this a bit more and indeed, setting `closingSingleTag` to `slash` will affect all self-closing tags in PostHTML... 

Is the ` />` mandatory in your use case? You could try [`replaceStrings`](https://maizzle.com/docs/transformers/replace-strings) to rewrite the end of the tag through I regex...

**Reply by bravecrayon** [reply]
@cossssmin Thanks for looking into it. It probably wasn't mandatory to have a `/>`, I can't quite remember now why I really wanted it.

I tried `replaceStrings` at one point but it didn't work — the `/` was taken out anyway.

This:
```
replaceStrings: {
  '<custom name="opencounter" type="tracking">': '<custom name="opencounter" type="asdf"/>'
}
```

Results in: 
```
<custom name="opencounter" type="asdf">
```
